cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -88,15 +115,19 @@ public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreExcep
   public void delete(NamespaceId namespaceId) throws IOException, ExploreException, SQLException {
     // TODO: CDAP-1581: Implement soft delete
     Location namespaceHome = namespacedLocationFactory.get(namespaceId.toId());
-    if (namespaceHome.exists()) {
-      if (!namespaceHome.delete(true)) {
-        throw new IOException(String.format("Error while deleting home directory '%s' for namespace '%s'",
-                                            namespaceHome, namespaceId.getNamespace()));
+    try {
+      if (Strings.isNullOrEmpty(namespaceQueryAdmin.get(namespaceId.toId()).getConfig().getRootDirectory())) {
```

Comment:
It could also be useful to move this to its own ``private boolean isCustomRootDirectory``, since its used twice